### PR TITLE
Eng 15149.rectify.calcite.branch

### PR DESCRIPTION
**Pull request by kaikaizi**
Fixed failed Jenkins tests, and merged from latest master branch (to fix another Jenkins error).

Relevant fixes are in `Adhoc.java` and `CalciteCompatibilityCheck.java`.

### CODE REVIEW

File: src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java [hunk 1]
```diff
@@ -63,6 +63,24 @@
     /** The list of all possible join orders, assembled by queueAllJoinOrders */
     private ArrayDeque<JoinNode> m_joinOrders = new ArrayDeque<>();
 
+    private static final Runtime RUN_TIME = Runtime.getRuntime();
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java]: RUNTIME

File: src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java [hunk 2]
```diff
@@ -516,7 +534,8 @@ private void generateInnerAccessPaths(BranchNode parentNode) {
     }
 
     /**
-     * generate all possible plans for the tree.
+     * generate all possible plans for the tree, or to the extend that further planning would drain JVM heap memory
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java]: to the extent

File: src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java [hunk 3]
```diff
@@ -516,7 +534,8 @@ private void generateInnerAccessPaths(BranchNode parentNode) {
...
+     * (at threshold of MAX_HEAP_MEMORY_USAGE_PCT% of available JVM heap memory)
```
Comment (yzhang1991) [inline src/frontend/org/voltdb/planner/SelectSubPlanAssembler.java]: used JVM heap memory

File: src/frontend/org/voltdb/plannerv2/guards/CalciteCompatibilityCheck.java [hunk 4]
```diff
@@ -51,18 +51,19 @@
 
     /**
      * Start the chained check to see if the SQL statement should be routed to Calcite.
+     * A query is permissible when at least one positive check in the chain passed (returns true);
+     * or when at least one negative check in the chain failed (returns false).
```
Comment (Chaozz) [inline src/frontend/org/voltdb/plannerv2/guards/CalciteCompatibilityCheck.java]: I am confused about the positive check and negative check.
`when at least one negative check in the chain failed (returns false).`

`NoLargeQuery` is a  negative check and return true if `! isLargeTempTableTarget` .
Which means if it is a large temp table target, the query is always permissible? I am thinking it is not permissible.

**Pull request merged**